src/compiler/checker.ts

```diff
@@ -8071,8 +8071,18 @@ namespace ts {
             return expression;
         }
 
+        function updateReferences(node: Identifier): void {
+            if (!isSourceFileADefinitionFile(node)) {
+                const symbol = getReferencedValueSymbol(node);
+                if (symbol) {
+                    symbol.hasReference = true;
+                }
+            }
+        }
+
         function checkIdentifier(node: Identifier): Type {
             const symbol = getResolvedSymbol(node);
+            updateReferences(node);

```

Comment:
now this will mark every value identifier even if they are not local/parameter. also given that you already have symbol on hands, why do you need to retrieve it again in `updateReferences`?